### PR TITLE
Add Doxygen docs to ip_grid_descriptor

### LINKED ISSUE #57: Add Doxygen to ip_grid_descriptor.f90

### PR DESCRIPTION
**Pull request by kgerheiser**
Fix #57

### CODE REVIEW

File: src/ip_grid_descriptor_mod.f90 [hunk 1]
```diff
@@ -1,3 +1,13 @@
+!> @file
+!! @brief Uses derived type grid descriptor objects to abstract away the raw Grib-1 and Grib-2 grid definitions.
+!! @author Kyle Gerheiser
+
+!> @brief Uses derived type grid descriptor objects to abstract away the raw Grib-1 and Grib-2 grid definitions.
+!! @detail
+!! An abstract ip_grid_descriptor class is subclassed by the Grib-1 and Grib-2 descriptor types which contain
+!! the raw integer grib arrays. A comparison operator to test whether two grid descriptors are the same.
```
Comment (edwardhartnett) [inline src/ip_grid_descriptor_mod.f90]: Can you properly line-break in your comments to keep them under 80 chars long?

Emacs has a command fill-paragraph which does this. Your editor no doubt also has this capability.

File: src/ip_grid_descriptor_mod.f90 [hunk 2]
```diff
@@ -1,3 +1,13 @@
+!> @file
+!! @brief Uses derived type grid descriptor objects to abstract away the raw Grib-1 and Grib-2 grid definitions.
+!! @author Kyle Gerheiser
```
Comment (edwardhartnett) [inline src/ip_grid_descriptor_mod.f90:4]: The file-level author should be the programmer who created this code file. Was that you?

File: src/ip_grid_descriptor_mod.f90 [hunk 3]
```diff
@@ -38,7 +55,11 @@ module ip_grid_descriptor_mod
   end interface init_descriptor
   
 contains
-  
+
+  !> @brief Initialize grib-1 descriptor from integer grid definition section
```
Comment (edwardhartnett) [inline src/ip_grid_descriptor_mod.f90]: Punctuate all sentences.

File: src/ip_grid_descriptor_mod.f90 [hunk 4]
```diff
@@ -49,6 +70,10 @@ function init_grib1_descriptor(gds) result(desc)
 
   end function init_grib1_descriptor
 
+  !> @brief Initialize grib-2 descriptor from integer grid definition template
```
Comment (edwardhartnett) [inline src/ip_grid_descriptor_mod.f90]: Punctuate. Also, for functions/subroutines you don't need to use the @breif tag, the first sentence will be used as the brief description, if AUTOBRIEF is turned on in Doxyfile, as it should be.

File: src/ip_grid_descriptor_mod.f90 [hunk 5]
```diff
@@ -49,6 +70,10 @@ function init_grib1_descriptor(gds) result(desc)
...
+  !! @param[in] gds Grib-2 GDT
```
Comment (edwardhartnett) [inline src/ip_grid_descriptor_mod.f90]: What is a GDT? Expand all abbreviations and acronyms.

File: src/ip_grid_descriptor_mod.f90 [hunk 6]
```diff
@@ -63,6 +88,11 @@ function init_grib2_descriptor(gdt_num, gdt_len, gdt_tmpl) result(desc)
     
   end function init_grib2_descriptor
 
+  !> @brief Test whether two grid descriptors are the same by comparing their type and value.
+  !! @param[in] grid1 An ip_grid_descriptor.
+  !! @param[in] grid2 Another ip_grid_descriptor.
+  !! @return True if the grids are the same, false if they are not.
+  !! @date July 2021
```
Comment (edwardhartnett) [inline src/ip_grid_descriptor_mod.f90:111]: We need an author tag for each function.

**Review by edwardhartnett — CHANGES_REQUESTED** [review]

File: src/ip_grid_descriptor_mod.f90 (hunk 2 above)
Comment (kgerheiser) [reply src/ip_grid_descriptor_mod.f90:4]: Yes

File: src/ip_grid_descriptor_mod.f90 [hunk 7]
```diff
@@ -1,3 +1,17 @@
+!> @file
+!! @brief Uses derived type grid descriptor objects to abstract away
+!! the raw Grib-1 and Grib-2 grid definitions.
+!! @author Kyle Gerheiser
+
+!> @brief Uses derived type grid descriptor objects to abstract away
+!!the raw Grib-1 and Grib-2 grid definitions.
+!! @detail
```
Comment (edwardhartnett) [inline src/ip_grid_descriptor_mod.f90]: I dont think we need or want the @detail tag. Instead, just leave an empty line to get a new paragraph. So:

```
!> @brief Uses derived type grid descriptor objects to abstract away
!! the raw Grib-1 and Grib-2 grid definitions.
!! 
!! An abstract ip_grid_descriptor class is subclassed by the
...
```

File: src/ip_grid_descriptor_mod.f90 [hunk 8]
```diff
@@ -85,6 +123,12 @@ logical function is_same_grid(grid1, grid2)
 
   end function is_same_grid
 
+  !> Test whether two grib1_descriptors are the same.
+  !! @param[in] self The grib1_descriptor which this routine was called on.
+  !! @param[in] grid_desc A grib1_descriptor to compare.
+  !! @return  True if the grids are the same, false if they are not.
+  !! @author Kyle Gerheiser
+  !! @date July 2021
```
Comment (edwardhartnett) [inline src/ip_grid_descriptor_mod.f90:141]: I usually add a blank line before and after params, and return tag. So:

```
 !> Test whether two grib1_descriptors are the same.
  !!
  !! @param[in] self The grib1_descriptor which this routine was called on.
  !! @param[in] grid_desc A grib1_descriptor to compare.
  !!
  !!
  !! @return  True if the grids are the same, false if they are not.
  !!
  !! @author Kyle Gerheiser
  !! @date July 2021
```

**Review by edwardhartnett — APPROVED** [review]

**Review by edwardhartnett — APPROVED** [review]

**Pull request merged**